src/views/edit-view.ts

```diff
@@ -279,36 +277,14 @@ export default class EditView extends View {
     }
   }
 
-  private dataUrlToArrayBuffer(dataURI: string): ArrayBuffer {
-    const byteString = atob(dataURI.split(',')[1]);
-    const ia = new Uint8Array(byteString.length);
-    for (let i = 0; i < byteString.length; i++) {
-        ia[i] = byteString.charCodeAt(i);
-    }
-    return ia.buffer as ArrayBuffer;
-  }
-
-  private async canvasToArrayBuffer(canvas: HTMLCanvasElement): Promise<ArrayBuffer> {
-    if (canvas.toBlob) {
-      const blob = await canvasToBlob(canvas, constants.IMAGE_TYPE);
-      const buffer = await blobToArrayBuffer(blob);
-      return buffer;
-    } else {
-      const dataURL = canvas.toDataURL(constants.IMAGE_TYPE);
-      const buffer = this.dataUrlToArrayBuffer(dataURL);
-      return buffer;
-    }
-  }
-
   private async save(): Promise<void> {
     this.draw();
-    const buffer = await this.canvasToArrayBuffer(this.imageShader.canvas);
+    const blob = await canvasToBlob(this.imageShader.canvas, constants.IMAGE_TYPE);
 
     if (this.currentRecord) {
```

Comment:
You may want to bail out earlier if `this.currentRecord` is null. We don't need to create `blob` if that's the case.